pusher-platform-android/src/main/kotlin/com/pusher/platform/AndroidDependencies.kt

```diff
@@ -21,9 +17,6 @@ data class AndroidDependencies(
 ) : PlatformDependencies {
 
     override val logger by lazy { AndroidLogger(threshold = LogLevel.DEBUG) }
-    override val scheduler: Scheduler by lazy { BackgroundScheduler() }
-    override val mainScheduler: MainThreadScheduler by lazy { ForegroundScheduler() }
     override val mediaTypeResolver: MediaTypeResolver by lazy { AndroidMediaTypeResolver() }
-    override val connectivityHelper: ConnectivityHelper by lazy { AndroidConnectivityHelper(context) }
```

Comment:
Schedulers are replaced with internal threading, no need to expose this. 
Using a `MainScheduler` can cause very unforeseeable consequences, like waking up the main thread after the app is in the background. The user probably has their own threading strategy where they can plug our SDK. 
ConnectivityHelper adds unnecessary requirements to the library, adding a need for having `ACCESS_NETWORK_STATE` enabled. We can gracefully recover from a network with incremental retries.

Reply:
Also, chances are that developers would defensively pass the call to the foreground (i.e. Using `observeOn()`) If not the app will crash pretty early.

Reply:
This is the explanation I was looking for at the top of the PR :+1: